src/dpp/sslclient.cpp

```diff
@@ -237,11 +249,11 @@ ssl_client::ssl_client(const std::string &_hostname, const std::string &_port, b
 	keepalive(reuse)
 {
 #ifndef WIN32
-	signal(SIGALRM, SIG_IGN);
-	signal(SIGHUP, SIG_IGN);
+	set_signal_handler(SIGALRM);
+	set_signal_handler(SIGHUP);
```

Comment:
I think SIGHUP should be ignored regardless? if it is captured we don't know if this will interrupt the library and break it? the signals will execute inside the libraries internal threads.